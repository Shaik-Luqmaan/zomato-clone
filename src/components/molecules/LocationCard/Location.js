import React from 'react';
import { makeStyles } from "@material-ui/core/styles";
import { Grid} from "@material-ui/core";
import Typography from "../../atoms/Typography/Typography";
import KeyboardArrowRightIcon from '@material-ui/icons/KeyboardArrowRight';

const useStyles = makeStyles({
    root: {
      border: "1px solid rgb(232,232,232)",
      height: "70px",
      maxWidth: "350px",
      borderRadius: "12px",
      padding: "20px",
      margin: "15px",
      boxShadow: "0 1.5px 2px rgb(100, 100, 100, 0.2)",
      color: "rgb(0,0,0,.8)",
     
      transition: "transform .2s",
      "&:hover": {
        boxShadow: "1px 2px 6px rgb(100, 100, 100, 0.2)",
      },
    arrow: {
      fontSize: "20px",
      opacity: .7,
    }, 
}});


const Location = ({placeName,placesCount,...props}) => {
    const classes = useStyles();
  return (
    <Grid container alignItems="center" className={classes.root} {...props}>
    <Grid item xs>
    <Typography {...props} variant="h6" data-testid="location">
      {placeName} ({placesCount})
    </Typography>
    </Grid>
    <Grid item>
    <KeyboardArrowRightIcon className={classes.arrow}/>
    </Grid>
  </Grid>
  )
}

export default Location
