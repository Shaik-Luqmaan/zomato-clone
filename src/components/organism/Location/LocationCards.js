import React from 'react'
import { makeStyles } from "@material-ui/core/styles";
import { Grid } from "@material-ui/core";
import Location from "../../molecules/LocationCard/Location"

const useStyles = makeStyles({
    root: {
      padding: "40px",
    },
  });

function LocationCards(props) {
    const classes = useStyles();
    // const { LocationDataList } = props;

    return (
        <Grid container className={classes.root} spacing={2}>
          {LocationDataList.map((value, index) => {
            return (
              <Grid item key={index} md={4} sm={6} xs={12} >
                  <Location {...value} ></Location>
              </Grid>
            );
          })}
        </Grid>
      );
}

const LocationDataList = [
    {
      placesCount: 540,
      placeName: "Tolichowki",
    },
    {
      placesCount: 451,
      placeName: "Mehdipatnam",
    },
    {
      placesCount: 187,
      placeName: "Asif Nagar",
    },
    {
      placesCount: 122,
      placeName: "Lakdikapool",
    },
    {
      placesCount: 48,
      placeName: "Nampally",
    },
    {
      placesCount: 363,
      placeName: "Jubilee Hills",
    },
    {
      placesCount: 584,
      placeName: "Gachibowli",
    },
    {
      placesCount: 264,
      placeName: "Hitech City",
    },
    {
      placesCount: 93,
      placeName: "Madhapur",
    },
      {
        placesCount: 363,
        placeName: "Banjara Hills",
      },
      {
        placesCount: 84,
        placeName: "Suncity",
      },
      {
        placesCount: 264,
        placeName: "Santosh Nagar",
      },
      {
        placesCount: 93,
        placeName: "Madhapur",
      },
      {
        placesCount: 1153,
        placeName: "Kukatpally",
      },
      {
        placesCount: 166,
        placeName: "Old City",
      },
    ]

export default LocationCards
