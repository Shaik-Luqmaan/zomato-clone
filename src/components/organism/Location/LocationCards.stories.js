import React from "react";
import LocationCards from "./LocationCards";

export default {
    title: "organisms/LocationCards",
    component: LocationCards,
}


const Template = ({...args})=>(
    <LocationCards {...args} />
);


export const Primary = Template.bind({});
Primary.args = {
   
}