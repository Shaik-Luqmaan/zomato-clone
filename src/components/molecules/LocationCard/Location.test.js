import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom"
import renderer from 'react-test-renderer';
import Location from './Location'

afterEach(cleanup);

it("renderer component correctly",()=>{
    const div = document.createElement('div');
    ReactDOM.render(< Location/>,div)
})

it("matches snapshot",()=>{
    const tree = renderer.create(<Location/>).toJSON();
    expect(tree).toMatchSnapshot();
})

it("renders component correctly", () => {
    const args = {
        placeName: "Tolichowki",
        placesCount: "500 places",
    };
    const { getByTestId } = render(<Location {...args} />);
    expect(getByTestId("location")).toHaveTextContent(
      args.placeName + " (" + args.placesCount + ")"
    );
  });

 