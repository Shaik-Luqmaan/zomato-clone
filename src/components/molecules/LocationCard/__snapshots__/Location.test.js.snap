// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-align-items-xs-center"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
  >
    <h6
      className="MuiTypography-root MuiTypography-h6"
      data-testid="location"
    >
       (
      )
    </h6>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
      />
    </svg>
  </div>
</div>
`;
